pallets/file-system/src/utils.rs

```diff
@@ -662,6 +855,13 @@ where
             let mut storage_request_metadata =
                 <StorageRequests<T>>::get(&file_key.0).ok_or(Error::<T>::StorageRequestNotFound)?;
 
+            if storage_request_metadata.bsps_confirmed == storage_request_metadata.bsps_required {
+                // Since BSPs need to race one another to confirm storage requests, it is entirely possible that a BSP confirms a storage request
+                // after the storage request has been fulfilled within the same block.
+                // TODO: Accumulate a list of storage requests that have been skipped for the BSP to confirm what it can.
+                continue;
+            }
```

Comment:
Good catch, let's set up a task to fix this asap.

Reply:
Done.